src/components/connections/Connection.ts

```diff
@@ -136,41 +129,45 @@ export class InfluxDBTreeDataProvider
     vscode.window.showInformationMessage("Success");
     return;
   }
-  private static async saveConn(
-    panel: vscode.WebviewPanel,
+
+  public static async SwitchConn(
     tree: InfluxDBTreeDataProvider,
-    message: Message
+    target: InfluxDBConnection
   ) {
     let connections = tree.context.globalState.get<{
       [key: string]: InfluxDBConnection;
     }>(InfluxDBConectionsKey);
-
     var hasConnection: Boolean = true;
     if (!connections) {
       hasConnection = false;
       connections = {};
     }
-
-    let id = message.connID;
-    if (id === "") {
-      id = uuidv1();
-    }
-    connections[id] = msg2Connection(message, id, true);
-
-    // flags other connections in case of default connections
-    if (message.connDefault && hasConnection) {
+    connections[target.id] = target;
+    // flags other connections in case of active connections
+    if (hasConnection) {
       for (const connID of Object.keys(connections)) {
-        if (id !== connID && connections[connID].isDefault) {
-          connections[connID].isDefault = false;
+        if (target.id !== connID && connections[connID].isActive) {
+          connections[connID].isActive = false;
         }
       }
     }
-    if (!hasConnection) {
-      // set the current connection, only if this is a new connection
-      Status.Current = connections[id];
-    }
+    Status.Current = connections[target.id];
     await tree.context.globalState.update(InfluxDBConectionsKey, connections);
     tree.refresh();
+  }
+
+  private static async saveConn(
+    panel: vscode.WebviewPanel,
+    tree: InfluxDBTreeDataProvider,
+    message: Message
+  ) {
+    let id = message.connID;
```

Comment:
```suggestion
    let id = message.connID || uuidv1();
```
this can be a little bit more concise this way